iotedgedev/deploymentmanifest.py

```diff
@@ -151,25 +157,46 @@ def validate_deployment_template(self):
     def validate_deployment_manifest(self):
         validation_success = True
         try:
-            validation_success = self._validate_deployment_manifest_schema()
+            if self.is_layered_deployment_schema():
+                validation_success = self._validate_deployment_manifest_schema()
             validation_success &= self._validate_create_options()
         except Exception as err:
             self.output.info("Unexpected error during deployment manifest validation, skip the validation. Error:%s" % err)
 
         return validation_success
 
+    def is_layered_deployment_schema(self):
+        return "content" in self.json
+
     @staticmethod
     def get_image_placeholder(module_name, is_debug=False):
         return "${{MODULES.{0}}}".format(module_name + ".debug" if is_debug else module_name)
 
+    def _get_module_content_split(self):
+        return DeploymentManifest.dot_to_json(self._get_module_content())
+
     def _get_module_content(self):
         if "modulesContent" in self.json:
             return self.json["modulesContent"]
         elif "moduleContent" in self.json:
             return self.json["moduleContent"]
+        elif "content" in self.json and "modulesContent" in self.json["content"]:
```

Comment:
>"content" in self.json [](http://example.com/codeflow?start=13&length=22)

Do you think that if the code uses is_layered_deployment_schema() instead here it will give more context to the reader?

Reply:
Could change it to:
```suggestion
        elif self.is_layered_deployment_schema() and "modulesContent" in self.json["content"]:
```